tests/test_useq_core_widgets.py

```diff
@@ -62,14 +67,136 @@ def test_core_connected_position_wdg(qtbot: QtBot, qapp) -> None:
     wdg._mmc.setZPosition(33)
     xyidx = pos_table.table().indexOf(pos_table._xy_btn_col)
     z_idx = pos_table.table().indexOf(pos_table._z_btn_col)
-    # i'm not sure why click() isn't working... but this is
+    # # i'm not sure why click() isn't working... but this is
     pos_table.table().cellWidget(0, xyidx).clicked.emit()
     pos_table.table().cellWidget(0, z_idx).clicked.emit()
     p0 = pos_table.value()[0]
     assert round(p0.x) == 11
     assert round(p0.y) == 22
     assert round(p0.z) == 33
 
+    wdg._mmc.waitForSystem()
     pos_table.move_to_selection.setChecked(True)
     pos_table.table().selectRow(0)
     pos_table._on_selection_change()
+
+
+@pytest.mark.parametrize("stage", ["XY", "Z"])
+def test_core_connected_position_wdg_cfg_loaded(
```

Comment:
these two tests are extremely similar... can you find a way to reduce a bit of the code duplication?

Reply:
yep :) I'm working on it